Source/Particles/MultiParticleContainer.cpp

```diff
@@ -461,3 +464,247 @@ MultiParticleContainer::doContinuousInjection() const
+                    // Copy particle from source to product: AoS
+                    p_product.id() = pid_product + ip;
+                    p_product.cpu() = cpuid;
+                    p_product.pos(0) = p_source.pos(0);
+                    p_product.pos(1) = p_source.pos(1);
+#if (AMREX_SPACEDIM == 3)
+                    p_product.pos(2) = p_source.pos(2);
+#endif
+                    // Copy particle from source to product: SoA
+                    for (int ia = 0; ia < PIdx::nattribs; ++ia) {
+                        attribs_product[ia][ip] = attribs_source[ia][is];
+                    }
+                    // Update xold etc. if boosted frame diagnostics required
+                    // for product species. Fill runtime attribs with a copy of 
+                    // current properties (xold = x etc.).
+                    if (do_boosted_product) {
+                        runtime_attribs_product[0][ip] = p_source.pos(0);
+                        runtime_attribs_product[1][ip] = p_source.pos(1);
+                        runtime_attribs_product[2][ip] = p_source.pos(2);
+                        runtime_attribs_product[3][ip] = runtime_uold_source[0][ip];
+                        runtime_attribs_product[4][ip] = runtime_uold_source[1][ip];
+                        runtime_attribs_product[5][ip] = runtime_uold_source[2][ip];
+                    }
+                }
+            }
+        );
+    }
+}
+
+void
+MultiParticleContainer::doFieldIonization ()
+{
+    BL_PROFILE("MPC::doFieldIonization");
+    // Loop over all species.
+    // Ionized particles in pc_source create particles in pc_product
+    for (auto& pc_source : allcontainers){
+    
+        // Skip if not ionizable
+        if (!pc_source->do_field_ionization){ continue; }
+
+        // Get product species
+        auto& pc_product = allcontainers[pc_source->ionization_product];
+
+        for (int lev = 0; lev <= pc_source->finestLevel(); ++lev){
+
+#ifdef _OPENMP
+            // Touch all tiles of source species in serial if runtime attribs 
+            for (MFIter mfi = pc_source->MakeMFIter(lev); mfi.isValid(); ++mfi) {
+                const int grid_id = mfi.index();
+                const int tile_id = mfi.LocalTileIndex();
+                pc_source->GetParticles(lev)[std::make_pair(grid_id,tile_id)];
+                if ( (pc_source->NumRuntimeRealComps()>0) || (pc_source->NumRuntimeIntComps()>0) ) {
+                    pc_source->DefineAndReturnParticleTile(lev, grid_id, tile_id);
+                }
+            }
+#endif
+            // Touch all tiles of product species in serial
+            for (MFIter mfi = pc_source->MakeMFIter(lev); mfi.isValid(); ++mfi) {
+                const int grid_id = mfi.index();
+                const int tile_id = mfi.LocalTileIndex();
+                pc_product->GetParticles(lev)[std::make_pair(grid_id,tile_id)];
+                pc_product->DefineAndReturnParticleTile(lev, grid_id, tile_id);
+            }
```

Comment:
@atmyers as discussed the other days, the two `MFIter` loops above do the following:

- `#ifdef _OPENMP` touch all tiles of the source particle container if runtime components are used;
- in any case, touch all tiles of the target particle container

where the source particle container is the ionizable species, and the target particle container is the species into which electrons are created.

Is this an appropriate way to do it?